### PR TITLE
Use generic names for RedHat and Debian-like clients

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

This PR replaces "ceos_minion" with "rhlike_minion" (RedHat-like minion) and "ubuntu_minion" with "dblike_minion" (Debian-like minion), so these minions become general.

This is in prevision of the replacement of CentOS by Rocky Linux.

It also does similar changes to tags, scopes, environment variables, file names, etc.

## Links

Fixes SUSE/spacewalk#18122

Related PRs:

* Sumaform: uyuni-project/sumaform#1122
* susemanager-ci: SUSE/susemanager-ci#595

Ports:

* 4.2: SUSE/spacewalk#18567
* 4.3: SUSE/spacewalk#18577

## Changelogs

- [x] No changelog needed

### CODE REVIEW

File: testsuite/config/cucumber.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ action_chains: --tags @scope_action_chains
 ansible: --tags @scope_ansible
 api: --tags @scope_api
 building_container_images: --tags @scope_building_container_images
-centos: --tags @centos_minion
+centos: --tags @rhlike_minion
```
Comment (ktsamis) [inline testsuite/config/cucumber.yml:6]: I don't see the usage of this scope in https://github.com/SUSE/susemanager-ci/pull/595/files. I see @scope_res could we have been using this wrong all along? Please verify this because you are not using @rhlike_minion anywhere now as well.

File: testsuite/documentation/idempotency.md [hunk 2]
```diff
@@ -25,9 +25,9 @@ As a standard status, we require that traditional clients and Salt minions have
 If you remove a traditional client or minion, **always re-add** the base channel, otherwise all package and patch tests will fail.
 
 
-## CentOS and Ubuntu minions
+## RedHat-like and Debian-like minions
 
-CentOS and Ubuntu clients are always registered as SSH salt minions by defaut.
+RedHat-like and Debian-like clients are always registered as Salt minions by defaut.
```
Comment (ktsamis) [inline testsuite/documentation/idempotency.md:30]: Why the change here? Are they salt ssh or salt minions? Are we missing ssh minions completely for RHLIKE and DBLIKE? Pretty much same question as above in the table.

File: testsuite/features/build_validation/init_clients/centos7_minion.feature [hunk 3]
```diff
@@ -4,29 +4,29 @@
 #  1) bootstrap a new CentOS 7 minion via salt-ssh
```
Comment (ktsamis) [inline testsuite/features/build_validation/init_clients/centos7_minion.feature:4]: This is not via salt-ssh, right? It's just webui without the ssh option checked. Please also change this

File: testsuite/features/build_validation/init_clients/centos8_minion.feature [hunk 4]
```diff
@@ -4,29 +4,29 @@
 #  1) bootstrap a new CentOS 8 minion via salt-ssh
```
Comment (ktsamis) [inline testsuite/features/build_validation/init_clients/centos8_minion.feature:4]: This is not via salt-ssh, just normal salt-minion, please change this as well.

File: testsuite/features/step_definitions/retail_steps.rb [hunk 5]
```diff
@@ -474,7 +474,7 @@ def compute_kiwi_profile_version(host)
   elsif host == 'sle15sp3_terminal'
     mac = $sle15sp3_terminal_mac
     mac = 'EE:EE:EE:00:00:06' if mac.nil?
-  elsif host.include? 'ubuntu'
+  elsif (host.include? 'dblike') || (host.include? 'debian11') || (host.include? 'ubuntu')
```
Comment (ktsamis) [inline testsuite/features/step_definitions/retail_steps.rb:477]: Here why we don't abstract dblike to ubuntu or debian? Why all these options?

File: testsuite/features/step_definitions/command_steps.rb [hunk 6]
```diff
@@ -484,15 +484,6 @@
   step %(I wait until file "/srv/tftpboot/pxelinux.cfg/default" contains "ONTIMEOUT local" on server)
 end
 
-When(/^I import the GPG keys for "([^"]*)"$/) do |host|
```
Comment (ktsamis) [inline testsuite/features/step_definitions/command_steps.rb:487]: This is not used anywhere?

File: testsuite/features/step_definitions/command_steps.rb [hunk 7]
```diff
@@ -175,7 +175,7 @@
     package.gsub! "suma", "uyuni"
   end
   node = get_target(host)
-  if host.include? 'ubuntu'
+  if db_host?(host)
```
Comment (ktsamis) [inline testsuite/features/step_definitions/command_steps.rb:178]: Where do you extract the db_host variable? I think this refactor is dangerous and might be syntactically wrong even. Instead of querying for whatever is present and saving it to host variable and comparing that with each value for each host, you are changing it to save each different value to a different variable named for each type of host. That could be dangerous if we try to use any of these variables in a place where they are not initialised.

File: testsuite/features/step_definitions/command_steps.rb [hunk 8]
```diff
@@ -894,11 +884,11 @@
 
 When(/^I install package(?:s)? "([^"]*)" on this "([^"]*)"((?: without error control)?)$/) do |package, host, error_control|
   node = get_target(host)
-  if host.include? 'ceos'
+  if rh_host?(host)
```
Comment (ktsamis) [inline testsuite/features/step_definitions/command_steps.rb:887]: Same question here, where do you set this variable?

File: testsuite/features/step_definitions/command_steps.rb [hunk 9]
```diff
@@ -960,9 +950,9 @@
 
 When(/^I wait until the package "(.*?)" has been cached on this "(.*?)"$/) do |pkg_name, host|
   node = get_target(host)
-  if host == 'sle_minion'
+  if suse_host?(host)
```
Comment (ktsamis) [inline testsuite/features/step_definitions/command_steps.rb:953]: Same question, where do you set this variable?

**Review by ktsamis — CHANGES_REQUESTED** [review]
See the comments/questions. Also please next time focus on one refactor per PR or at least contain the changes in one commit at a time. This was very hard to review all at once in once commit.

File: testsuite/documentation/cucumber-steps.md [hunk 10]
```diff
@@ -38,9 +38,9 @@ Possible values are currently:
 | SLES traditional client | ```$client``` | ```$CLIENT``` | ```"sle_client"``` | ```"client"``` |
 | SLES Salt minion | ```$minion``` | ```$MINION``` | ```"sle_minion"``` | ```"minion"``` |
 | SLES Docker and Kiwi build host | ```$build_host``` | ```$BUILD_HOST``` | ```"build_host"``` | ```"minion"``` |
-| SLES Salt SSH minion | ```$ssh_minion``` | ```$SSHMINION``` | ```"ssh_minion"``` | ```"minion"``` |
-| CentOS Salt minion | ```$ceos_minion``` | ```$CENTOSMINION``` | ```"ceos_minion"``` | ```"minion"``` |
-| Ubuntu Salt minion | ```$ubuntu_minion``` | ```$UBUNTUMINION``` | ```"ubuntu_minion"``` | ```"minion"``` |
+| SLES Salt SSH minion | ```$ssh_minion``` | ```$SSH_MINION``` | ```"ssh_minion"``` | ```"minion"``` |
+| RedHat-like Salt minion | ```$rhlike_minion``` | ```$RHLIKE_MINION``` | ```"rhlike_minion"``` | ```"minion"``` |
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply testsuite/documentation/cucumber-steps.md:42]: No.

Sumaform provides only one VM for each, and that means only one environment variable.

The "normal CI" then converts them to SSH for the SSH tests, and then back to normal Salt.

The BV test suite does differently : each flavour is one different VM. That is much simpler and cleaner, but also implies loads of VMs, which is not something you can do without a heavy infrastructure, and certainly not on a workstation.

File: testsuite/config/cucumber.yml (hunk 1 above)
Comment (Bischoff) [reply testsuite/config/cucumber.yml:6]: Good catch :smile: . I caught it too. I was tempted to remove it, but there might be this reason for this weirdness:

There is no `scope_centos` defined. They probably did not do it because it would have been an exact duplicate of `@centos_minion` (not in the semantics, but for the places where it appears).

If the user wants to toggle the CentOS scope on/off, that's the only way to acheive it.

It's only an educated guess. Maybe Jordi can confirm.

File: testsuite/features/build_validation/init_clients/centos8_minion.feature (hunk 4 above)
Comment (Bischoff) [reply testsuite/features/build_validation/init_clients/centos8_minion.feature:4]: Correct.

File: testsuite/features/step_definitions/retail_steps.rb (hunk 5 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/retail_steps.rb:477]: CI test suite : generic `dblike`
BV test suite : specific `ubuntu1804`, `ubuntu2004`, `ubuntu2204`, `debian9`, `debian10`, `debian11`

Notice that `debian9` and `debian10` use normal `eth1`instead of `en4`...

File: testsuite/features/step_definitions/command_steps.rb (hunk 6 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:487]: No. Dead code.

File: testsuite/features/step_definitions/command_steps.rb (hunk 7 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:178]: it's not a variable, it's a function, defined in `commonlib.rb`.

File: testsuite/features/step_definitions/command_steps.rb (hunk 8 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:887]: it's not a variable, and it's in `commonlib.rb`

File: testsuite/features/step_definitions/command_steps.rb (hunk 9 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:953]: and same answer.

File: testsuite/documentation/cucumber-steps.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ktsamis) [reply testsuite/documentation/cucumber-steps.md:42]: OK so we reuse the VM, all good

File: testsuite/config/cucumber.yml (hunk 1 above)
Comment (ktsamis) [reply testsuite/config/cucumber.yml:6]: I don't think I see the mapping of the scope to the centos_minion tag, which are different things in scope. Maybe @jordimassaguerpla can take a look at this?

File: testsuite/features/build_validation/init_clients/centos7_minion.feature (hunk 3 above)
Comment (ktsamis) [reply testsuite/features/build_validation/init_clients/centos7_minion.feature]: Thanks for fixing this extra thing

File: testsuite/features/step_definitions/retail_steps.rb (hunk 5 above)
Comment (ktsamis) [reply testsuite/features/step_definitions/retail_steps.rb:477]: All good I think here, it's still the same

**Review by ktsamis — APPROVED** [review]
Except the scope question which is for Jordi (which the question is out of scope for this PR anyway) it looks good. Thanks for making this beast manageable Eric.

**Comment by Bischoff** [comment]
> See the comments/questions. Also please next time focus on one refactor per PR or at least contain the changes in one commit at a time. This was very hard to review all at once in once commit.

Right. Refactored into 3 commits.

**Comment by srbarrios** [comment]
Nitpick @Bischoff 
Could we use "deb" instead of "db" in the renamings?
Just to avoid unnecessary missunderstandings.

File: testsuite/features/support/twopence_init.rb [hunk 11]
```diff
@@ -331,22 +332,19 @@ def client_public_ip(host)
   end
 
   interface = case host
-              when /^sle/, /^opensuse/, /^ssh/, /^ceos/, /^debian9/, /^debian10/, 'server', 'proxy', 'build_host'
-                'eth0'
-              when /^debian11/, /^ubuntu/
+              when /^dblike/, /^debian11/, /^ubuntu/
```
Comment (srbarrios) [inline testsuite/features/support/twopence_init.rb:335]: I think we should keep debian10 in that list. We still testing it in a BV right? 🤔

File: testsuite/features/support/constants.rb [hunk 12]
```diff
@@ -150,11 +150,11 @@
...
-                           'ceos7_client' => 'RHEL x86_64 Server 7',
-                           'ceos7_minion' => 'RHEL x86_64 Server 7',
-                           'ceos7_ssh_minion' => 'RHEL x86_64 Server 7',
-                           'ceos8_minion' => 'no-appstream-result-RHEL8-Pool for x86_64',
-                           'ceos8_ssh_minion' => 'no-appstream-result-RHEL8-Pool for x86_64',
+                           'centos7_client' => 'RHEL x86_64 Server 7',
+                           'centos7_minion' => 'RHEL x86_64 Server 7',
+                           'centos7_ssh_minion' => 'RHEL x86_64 Server 7',
+                           'centos8_minion' => 'no-appstream-result-RHEL8-Pool for x86_64',
+                           'centos8_ssh_minion' => 'no-appstream-result-RHEL8-Pool for x86_64',
```
Comment (srbarrios) [inline testsuite/features/support/constants.rb:157]: :D You know what I think about this piece of code changes in that PR.

File: testsuite/features/support/commonlib.rb [hunk 13]
```diff
@@ -176,6 +176,18 @@ def get_client_type(name)
   end
 end
 
+def suse_host?(name)
+  (name.include? 'sle') || (name.include? 'opensuse')
+end
+
+def rh_host?(name)
```
Comment (srbarrios) [inline testsuite/features/support/commonlib.rb:183]: We might need more soon. Like other RH like distros, Oracle Linux, Amazon Linux, and so on.

Comment (Bischoff) [reply testsuite/features/support/commonlib.rb:183]: That's the exact motivation for this PR.

**Comment by Bischoff** [comment]
> Nitpick @Bischoff Could we use "deb" instead of "db" in the renamings? Just to avoid unnecessary misunderstandings.

I hesitated a lot.

* "db_like" is short like "rh_like"
* but as you say it's less understandable

I'll change it, but I will need to change in sumaform as well, then.

File: testsuite/features/support/constants.rb (hunk 12 above)
Comment (Bischoff) [reply testsuite/features/support/constants.rb:157]: Let's kill `ceos`, it's not readable. It's same issue ad `dblike` versus `deblike`.

**Comment by Bischoff** [comment]
> Nitpick @Bischoff Could we use "deb" instead of "db" in the renamings? Just to avoid unnecessary misunderstandings.

Yes. I was very tempted myself to do that.

File: testsuite/config/cucumber.yml [hunk 14]
```diff
@@ -35,7 +35,7 @@ subscription_matching: --tags @scope_subscription_matching
 sumatoolbox: --tags @scope_sumatoolbox
 tomcat: --tags @scope_tomcat
 traditional_client: --tags @scope_traditional_client
-ubuntu: --tags @scope_ubuntu
+ubuntu: --tags @scope_deblike
```
Comment (vandabarata) [inline testsuite/config/cucumber.yml:38]: Just a reminder to not forget to change the scopes on the CI files themselves too

File: testsuite/documentation/optional.md [hunk 15]
```diff
@@ -119,46 +119,48 @@ Inside of the testsuite, the scenarios that are tagged with
...
-### Testing with an Ubuntu minion
+### Testing with a Debian-like minion
 
-Using an Ubuntu minion with the testsuite is not mandatory.
+Using an Debian-like minion (Debian, Ubuntu, ...) with the test suite
+is not mandatory.
 
-If you do not want an Ubuntu minion, do not define `UBUNTUMINION` environment
-variable before you run the testsuite. That's all.
+If you do not want an Debian-like minion, do not define `DEBLIKE_MINION`
```
Comment (vandabarata) [inline testsuite/documentation/optional.md:149]:
```suggestion
If you do not want a Debian-like minion, do not define `DEBLIKE_MINION`
```
I know it's a tiny nitpick but...

File: testsuite/documentation/optional.md [hunk 16]
```diff
@@ -119,46 +119,48 @@ Inside of the testsuite, the scenarios that are tagged with
...
+environment variable before you run the test suite. That's all.
 
-If you want an Ubuntu minion, make this variable point to the machine that
-will be the Ubuntu minion:
+If you want an Debian-like minion, make this variable point to the machine
```
Comment (vandabarata) [inline testsuite/documentation/optional.md:152]:
```suggestion
If you want a Debian-like minion, make this variable point to the machine
```

File: testsuite/features/init_clients/min_dblike_salt.feature [hunk 17]
```diff
@@ -1,47 +1,47 @@
...
-@ubuntu_minion
-Feature: Bootstrap an Ubuntu minion and do some basic operations on it
+@deblike_minion
+Feature: Bootstrap an Debian-like minion and do some basic operations on it
 
   Scenario: Log in as admin user
     Given I am authorized for the "Admin" section
 
-  Scenario: Bootstrap an Ubuntu minion
+  Scenario: Bootstrap an Debian-like minion
```
Comment (vandabarata) [inline testsuite/features/init_clients/min_dblike_salt.feature:13]:
```suggestion
  Scenario: Bootstrap a Debian-like minion
```

File: testsuite/features/step_definitions/retail_steps.rb [hunk 18]
```diff
@@ -474,7 +474,7 @@ def compute_kiwi_profile_version(host)
   elsif host == 'sle15sp3_terminal'
     mac = $sle15sp3_terminal_mac
     mac = 'EE:EE:EE:00:00:06' if mac.nil?
-  elsif host.include? 'ubuntu'
+  elsif (host.include? 'deblike') || (host.include? 'debian11') || (host.include? 'ubuntu')
```
Comment (vandabarata) [inline testsuite/features/step_definitions/retail_steps.rb:477]: can't you use the `db_host?(host)` function here like you've been using it for the rest of the code? :thinking:

File: testsuite/features/support/commonlib.rb [hunk 19]
```diff
@@ -176,6 +176,18 @@ def get_client_type(name)
...
+  (name.include? 'rhlike') || (name.include? 'centos') || (name.include? 'alma') || (name.include? 'rocky')
+end
+
+def db_host?(name)
```
Comment (vandabarata) [inline testsuite/features/support/commonlib.rb:187]: this naming can get confusing as `db` can be seen as database instead of debian - would it be too much of a hassle to change it to
```suggestion
def ddb_host?(name)
```
instead? :grimacing:

File: testsuite/config/cucumber.yml (hunk 14 above)
Comment (Bischoff) [reply testsuite/config/cucumber.yml:38]: As said in this PR's description, it is addressed in:

SUSE/susemanager-ci#595

File: testsuite/documentation/optional.md (hunk 15 above)
Comment (Bischoff) [reply testsuite/documentation/optional.md:149]: oops, thanks, doing it

Comment (Bischoff) [reply testsuite/documentation/optional.md]: Done, there were 7 occurrences :-(

File: testsuite/documentation/optional.md (hunk 16 above)
Comment (Bischoff) [reply testsuite/documentation/optional.md]: See above

File: testsuite/features/init_clients/min_dblike_salt.feature (hunk 17 above)
Comment (Bischoff) [reply testsuite/features/init_clients/min_dblike_salt.feature]: See above

File: testsuite/features/step_definitions/retail_steps.rb (hunk 18 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/retail_steps.rb:477]: No, as you can see, `debian11` does not work like `debian9` nor `debian10`.

They changed the interface naming conventions from one version to the other.

File: testsuite/features/support/commonlib.rb (hunk 19 above)
Comment (Bischoff) [reply testsuite/features/support/commonlib.rb:187]: Thanks, I missed that one when I followed Oscar's suggestion to change `db` into `deb`.

**Review by vandabarata — APPROVED** [review]

File: testsuite/config/cucumber.yml (hunk 14 above)
Comment (vandabarata) [reply testsuite/config/cucumber.yml:38]: I missed it in the description, sorry :see_no_evil:

File: testsuite/features/secondary/srv_monitoring.feature [hunk 20]
```diff
@@ -1,8 +1,10 @@
 # Copyright (c) 2019-2022 SUSE LLC
 # Licensed under the terms of the MIT license.
+#
 # This feature is a dependency for:
 # - features/secondary/min_monitoring.feature : If this feature fails could let monitoring feature disabled for SLE minion
-# - features/secondary/min_centos_monitoring.feature : If this feature fails could let monitoring feature disabled for CentOS minion
+# - features/secondary/min_rhlike_monitoring.feature : If this feature fails could let monitoring feature disabled for RedHat-like minion
```
Comment (nodeg) [inline testsuite/features/secondary/srv_monitoring.feature:6]:
```suggestion
# - features/secondary/min_rhlike_monitoring.feature : If this feature fails, it could let the monitoring feature disabled for RedHat-like minion
```

Comment (nodeg) [reply testsuite/features/secondary/srv_monitoring.feature:6]: Same for line 5 but you did not touch that one.

**Review by nodeg — APPROVED** [review]
I left some small suggestions, otherwise LGTM.

File: testsuite/features/build_validation/init_clients/centos7_client.feature [hunk 21]
```diff
@@ -1,28 +1,28 @@
...
+    When I perform a full salt minion cleanup on "centos7_client"
 
   Scenario: Prepare a CentOS 7 traditional client
-    When I enable repository "CentOS-Base tools_pool_repo" on this "ceos7_client" without error control
-    And I bootstrap traditional client "ceos7_client" using bootstrap script with activation key "1-ceos7_client_key" from the proxy
-    And I install the traditional stack utils on "ceos7_client"
-    And I run "mgr-actions-control --enable-all" on "ceos7_client"
-    Then I should see "ceos7_client" via spacecmd
+    When I enable repository "CentOS-Base tools_pool_repo" on this "centos7_client" without error control
+    And I bootstrap traditional client "centos7_client" using bootstrap script with activation key "1-centos7_client_key" from the proxy
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply testsuite/features/build_validation/init_clients/centos7_client.feature:12]: This is created from template:
```
  Scenario: Create an activation key with the channel and child channels for a <client>
    When I am logged in API as user "admin" and password "admin"
    And I create an activation key including custom channels for "<client>" via API
    And I logout from API
```
`<client>` should be replaced with `centos7`.

File: testsuite/features/secondary/srv_monitoring.feature (hunk 20 above)
Comment (Bischoff) [reply testsuite/features/secondary/srv_monitoring.feature:6]: Thank you, fixing.

File: testsuite/features/build_validation/init_clients/centos7_client.feature (hunk 21 above)
[the remark this answers is not in the document]
Comment (nodeg) [reply testsuite/features/build_validation/init_clients/centos7_client.feature:12]: Ok, then everything is fine. Thanks for the clarification.

File: testsuite/README.md [hunk 22]
```diff
@@ -28,7 +28,7 @@ The **secondary features** can be run any number of times, and the order is not
 
 ## Optional components
 
-To know how to test with or without optional components like a proxy, a CentOS minion or a SSH minion, look at the [optional components instructions](documentation/optional.md).
+To know how to test with or without optional components like a proxy, a RedHat-like minion or a SSH minion, look at the [optional components instructions](documentation/optional.md).
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply testsuite/README.md]: Fixed, thanks.

File: testsuite/features/secondary/srv_monitoring.feature [hunk 23]
```diff
@@ -1,10 +1,12 @@
 # Copyright (c) 2019-2022 SUSE LLC
 # Licensed under the terms of the MIT license.
+#
 # This feature is a dependency for:
-# - features/secondary/min_monitoring.feature : If this feature fails could let monitoring feature disabled for SLE minion
-# - features/secondary/min_centos_monitoring.feature : If this feature fails could let monitoring feature disabled for CentOS minion
+# - features/secondary/min_monitoring.feature: if this feature fails, it could let monitoring feature disabled for SLE minion
+# - features/secondary/min_rhlike_monitoring.feature: if this feature fails, it could let monitoring feature disabled for Red Hat-like minion
```
Comment (nodeg) [inline testsuite/features/secondary/srv_monitoring.feature:6]:
```suggestion
# - features/secondary/min_monitoring.feature: if this feature fails, it could let the monitoring feature disabled for SLE minion
# - features/secondary/min_rhlike_monitoring.feature: if this feature fails, it could let the monitoring feature disabled for Red Hat-like minion
```

**Comment by juliogonzalez** [comment]
Since we are trying to make the testsuite stable for Uyuni 2022.07 and SUSE Manager 4.3.1, I need QE to decide if we really want to merge this PR now.

If so, readd the label and ping me on slack.

**Comment by srbarrios** [comment]
> Since we are trying to make the testsuite stable for Uyuni 2022.07 and SUSE Manager 4.3.1, I need QE to decide if we really want to merge this PR now.
> 
> If so, readd the label and ping me on slack.

We agreed to don't merge it until you submit. So, until the latest added content it's stable in the tests.

File: testsuite/documentation/optional.md [hunk 24]
```diff
@@ -119,46 +119,48 @@ Inside of the testsuite, the scenarios that are tagged with
...
+### Testing with a Red Hat-like minion
 
-Using a CentOS minion with the testsuite is not mandatory.
+Using a Red Hat-like minion (CentOS, Alma, Rocky, ...) with the test suite
+is not mandatory.
 
-If you do not want a CentOS minion, do not define `CENTOSMINION` environment
-variable before you run the testsuite. That's all.
+If you do not want a Red Hat-like minion, do not define `RHLIKE_MINION`
+environment variable before you run the test suite. That's all.
```
Comment (nodeg) [inline testsuite/documentation/optional.md:128]:
```suggestion
If you do not want a Red Hat-like minion, do not define a `RHLIKE_MINION`
environment variable before you run the test suite. That's all.
```

File: testsuite/documentation/optional.md [hunk 25]
```diff
@@ -119,46 +119,48 @@ Inside of the testsuite, the scenarios that are tagged with
...
+### Testing with a Debian-like minion
 
-Using an Ubuntu minion with the testsuite is not mandatory.
+Using a Debian-like minion (Debian, Ubuntu, ...) with the test suite
+is not mandatory.
 
-If you do not want an Ubuntu minion, do not define `UBUNTUMINION` environment
-variable before you run the testsuite. That's all.
+If you do not want a Debian-like minion, do not define `DEBLIKE_MINION`
+environment variable before you run the test suite. That's all.
```
Comment (nodeg) [inline testsuite/documentation/optional.md:150]:
```suggestion
If you do not want a Debian-like minion, do not define a `DEBLIKE_MINION`
environment variable before you run the test suite. That's all.
```

File: testsuite/features/secondary/min_rhlike_remote_command.feature [hunk 26]
```diff
@@ -1,23 +1,23 @@
-# Copyright (c) 2021 SUSE LLC
+# Copyright (c) 2021-2022 SUSE LLC
 # Licensed under the terms of the MIT license.
 
 @scope_res
-Feature: Remote command on CentOS Salt minion
-  In order to manage a CentOS Salt minion
+@rhlike_minion
+Feature: Remote command on Red Hat-like Salt minion
```
Comment (nodeg) [inline testsuite/features/secondary/min_rhlike_remote_command.feature:6]:
```suggestion
Feature: Remote command on a Red Hat-like Salt minion
```
or
```suggestion
Feature: Remote command on the Red Hat-like Salt minion
```

File: testsuite/features/secondary/srv_monitoring.feature [hunk 27]
```diff
@@ -1,10 +1,12 @@
 # Copyright (c) 2019-2022 SUSE LLC
 # Licensed under the terms of the MIT license.
+#
 # This feature is a dependency for:
-# - features/secondary/min_monitoring.feature : If this feature fails could let monitoring feature disabled for SLE minion
-# - features/secondary/min_centos_monitoring.feature : If this feature fails could let monitoring feature disabled for CentOS minion
+# - features/secondary/min_monitoring.feature: if this feature fails, it could let the monitoring feature disabled for SLE minion
+# - features/secondary/min_rhlike_monitoring.feature: if this feature fails, it could let the monitoring feature disabled for Red Hat-like minion
```
Comment (nodeg) [inline testsuite/features/secondary/srv_monitoring.feature:6]:
```suggestion
# - features/secondary/min_monitoring.feature: if this feature fails, it could let the monitoring feature disabled for the SLE minion
# - features/secondary/min_rhlike_monitoring.feature: if this feature fails, it could let the monitoring feature disabled for the Red Hat-like minion
```

File: testsuite/features/step_definitions/common_steps.rb [hunk 28]
```diff
@@ -855,7 +855,7 @@
   node = get_target(client)
   if client.include? 'sle'
     node.run('zypper --non-interactive install wget', timeout: 500)
-  else # As Ubuntu has no support, must be CentOS/SLES_ES
+  else # As Debian-like has no support for trad clients, must be Red Hat-like
```
Comment (nodeg) [inline testsuite/features/step_definitions/common_steps.rb:858]:
```suggestion
  else # As Debian-like has no support for traditional clients, it must be Red Hat-like
```

**Review by nodeg — APPROVED** [review]
Some nitpicks, otherwise LGTM.

File: testsuite/documentation/optional.md (hunk 24 above)
Comment (Bischoff) [reply testsuite/documentation/optional.md:128]: No, this variable is definite and unique, I think it's correct (and better) without the undefinite article.

File: testsuite/documentation/optional.md (hunk 25 above)
Comment (Bischoff) [reply testsuite/documentation/optional.md:150]: Same

File: testsuite/features/secondary/min_rhlike_remote_command.feature (hunk 26 above)
Comment (Bischoff) [reply testsuite/features/secondary/min_rhlike_remote_command.feature:6]: yes, thanks

File: testsuite/features/secondary/srv_monitoring.feature (hunk 27 above)
Comment (Bischoff) [reply testsuite/features/secondary/srv_monitoring.feature:6]: yes too

File: testsuite/features/step_definitions/common_steps.rb (hunk 28 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb:858]: yup, we speak well too often in "telegraphic style"

File: testsuite/features/secondary/min_rhlike_remote_command.feature (hunk 26 above)
Comment (Bischoff) [reply testsuite/features/secondary/min_rhlike_remote_command.feature:6]: done

File: testsuite/features/step_definitions/common_steps.rb (hunk 28 above)
Comment (srbarrios) [reply testsuite/features/step_definitions/common_steps.rb:858]: I'm very good with this style!

File: testsuite/features/secondary/srv_monitoring.feature (hunk 27 above)
Comment (Bischoff) [reply testsuite/features/secondary/srv_monitoring.feature:6]: done too

File: testsuite/features/step_definitions/common_steps.rb (hunk 28 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb:858]: LOL

Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb:858]: done

File: testsuite/documentation/optional.md (hunk 24 above)
Comment (nodeg) [reply testsuite/documentation/optional.md:128]: Alright.
